### ISSUE TITLE
OIDC self signed certificate

### ISSUE THREAD

**Issue by RGanor** [PowerDNS-Admin/PowerDNS-Admin#699]
I'm having this issue with self signed certificates for my oidc, I tried to put the CA on every known location on alpine but it keep ignore it. I started debugging the code until i found out that `certifi` package using its own custom CA file, my work around for that is adding this to docker-compose.yml:
```
volumes:
      - "/etc/pki/ca-trust/source/anchors/cacert.pem:/usr/lib/python3.8/site-packages/certifi/cacert.pem"
```
Is there any, more legitimate, way to use self signed certificates rather than changing the cacert in python site-package, and anyway, i think we should at least mention that in the wiki so more people can easily overcome this problem.

**Comment by ngoduykhanh** [comment]
As I remember I faced this issue during some tests in the past. Unless I manually edit the authlib code, I have to use a trusted certificate for OpenID server.

**Comment by maximumG** [comment]
I am using the environment variable `AUTHLIB_INSECURE_TRANSPORT` set to `true` in order to instruct the authlib NOT to check the OIDC provider certificate

**Comment by git-noise** [comment]
Hello,
I am using something slightly different. Some packages will populate or override their SSL trust store from specific ENV variables.
As such variables like REQUESTS_CA_BUNDLE or CURL_CA_BUNDLE may allow to pass a specific CA bundle without altering the code.
So trusting the cert on the host machine - so that the ca-certificates.crt bundle file contains the self-signed CA - and then adding the following into the docker-compose file may help:
```
volumes:
  - /etc/ssl/certs/ca-certificates.crt:/etc/tls/cacert.crt
environment:
  - REQUESTS_CA_BUNDLE=/etc/tls/cacert.crt
```
Otherwise I also believe that some part of authlib functions can handle parameters passed down to the request function, but I could not find any clear documentation and it would require some code changes I guess.

**Comment by ttinkr** [comment]
I tried every mentioned solution to this, but unfortunately without luck.
Is there any other way to achieve this or maybe a hint someone can give me?

I mounted a cacert.pem to /usr/lib/python3.8/site-packages/certifi/ with our own CA, but i still see the error.
Even when I set AUTHLIB_INSECURE_TRANSPORT it doesn't work, but maybe because I run this in a container and the variable is not in the legal_envvars?

Thanks!
